sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+
+namespace Azure.Messaging.ServiceBus.Amqp
+{
+    internal static class AmqpMessageExtensions
+    {
+        private static AmqpMessage ToAmqpDataMessage(this ServiceBusMessage message, bool viaBody)
+        {
+            IEnumerable<ReadOnlyMemory<byte>> bodyBytes = viaBody ? new[] { message.Body.AsBytes() } : message.GetAmqpDataBody();
```

Comment:
Do we still need `viaBody` now that `GetAmqpDataBody` works for messages created via SBM(ROM<bytes>) ctor?

Reply:
Yes. So, I can add more comments to explain, but this is related to that 'magic sauce' I mentioned above. `viaBody` is false for Data body type and true for Unspecified. Unspecified occurs when the user hasn't used the extension methods to create the message. `GetAmqpDataBody` on ServiceBusMessage only gets the information from Data, which will be `null` when the body type is Unspecified. Thus, `viaBody` says that the information is stored in Body versus being stored in Data. I believe you are thinking of the `GetAmqpDataBody` on  ServiceBusReceivedMessage, which functions differently. They function differently because the body type for ServiceBusMessage is explicit where the body type for ServiceBusReceivedMessage is implicit, based on the situation described in my 'magic sauce' comments.

Reply:
But since GetAmqpDataBody is changing to work for either implicit or explicit data messages, I am wondering if this branching can go away.

Reply:
This has been simplified and viaBody is removed. 😄